ros2caret/verb/record.py

```diff
@@ -221,7 +226,16 @@ def main(self, *, args):
             raise ValueError('--subbuffer-size-kernel value must be power of two.')
         init_args['subbuffer_size_kernel'] = args.subbuffer_size_kernel
         init_args['immediate'] = args.immediate
-        init(**init_args)
+
+        for i in range(args.init_retry_num):
+            init_result = init(**init_args)
+            if init_result:
+                break
+            print(f'Failed to init LTTng. retry {i} / {args.init_retry_num}')
+            time.sleep(1)
+        if not init_result:
+            print('Failed to init LTTng.')
+            exit(0)
```

Comment:
If init_retry_num is an integer less than or equal to 0, 'init_result' will be referenced before assignment.
```
UnboundLocalError: local variable 'init_result' referenced before assignment
```
This problem can be avoided with the following simple code change.

```suggestion
        for i in range(args.init_retry_num):
            init_result = init(**init_args)
            if init_result:
                break
            print(f'Failed to init LTTng. retry {i} / {args.init_retry_num}')
            time.sleep(1)
        else:
            print('Failed to init LTTng.')
            exit(0)
```

Reply:
Thank you for the suggestion. I didn't know this coding way!!
I checked it works and commit your suggestion.